src/vm/analysis/type_checker/mod.rs

```diff
@@ -201,12 +204,23 @@ impl <'a, 'b> TypeChecker <'a, 'b> {
                 self.type_check(&exp, &local_context)?;
             }
         }
-
         Ok(())
     }
 
     // Type check an expression, with an expected_type that should _admit_ the expression.
     pub fn type_check_expects(&mut self, expr: &SymbolicExpression, context: &TypingContext, expected_type: &TypeSignature) -> TypeResult {
+        
+        match (&expr.expr, expected_type) {
+            (LiteralValue(Value::Principal(PrincipalData::Contract(ref contract_identifier))), TypeSignature::TraitReferenceType(trait_identifier)) => {
+                let traits = self.db.get_implemented_traits(&contract_identifier).unwrap();
+                println!("===> HERE {:?}", traits);
```

Comment:
Please remove this ;)

Reply:
#cavemandebugging 
Addressed with https://github.com/blockstack/blockstack-core/pull/1246/commits/fee30e832e02d44f5c00e61d9cfa264a4229700b

Reply:
If it's good enough for Brian Kernighan, it's good enough for me :grin: